include/rdma/fi_domain.h

```diff
@@ -103,6 +103,7 @@ struct fi_mr_attr {
 	uint64_t		offset;
 	uint64_t		requested_key;
 	void			*context;
+	size_t			auth_key_index;
```

Comment:
I'd prefer not to see the addition of methods to register memory against endpoints as a commit in this PR, but rather opened as a separate issue.  This seems to have a lot of implications for client applications that needs to be discussed.  Tracking such a discussion would be easier done via a separate issue/PR.